src/robot/errors.py

```diff
@@ -53,6 +53,13 @@ class DataError(RobotError):
     this exception with care.
     """
 
+class VariableError(DataError):
+    """Used when variable syntax or evaluation is invalid.
```

Comment:
Isn't this error only used when variable doesn't exist? I'll update docs myself.